kano/network.py

```diff
@@ -72,22 +72,29 @@ def parseRawData(rawdata):
             cellDataL = []
             #currentCell = None
             for s in rawdatas:
-                # skip empty lines
-                if not s.strip():
-                    continue
-                # If new cell:
-                if s.lstrip().startswith("Cell "):
-                    # log.debug("parseRawData: new cell")
-                    cellDataL.append([])
-                if len(cellDataL) > 0 and len(s) > 0:
-                    cellDataL[len(cellDataL) - 1].append(s)
+                try:
+                    # skip empty lines
+                    if not s.strip():
+                        continue
+                    # If new cell:
+                    if s.lstrip().startswith("Cell "):
+                        # log.debug("parseRawData: new cell")
+                        cellDataL.append([])
+                    if len(cellDataL) > 0 and len(s) > 0:
+                        cellDataL[len(cellDataL) - 1].append(s)
+                except Exception as e:
+                    logger.error('unexpected error occurred while looping rawdatas {}'
+                                 .format(rawdata), exception=e)
 
             # Data is separated by cells, now we'll parse each cell's data
             parsedCellData = {}
             for s in cellDataL:
-                cellNumber, cellData = parseCellData("\n".join(s))
-                parsedCellData[cellNumber] = cellData
-            #log.debug("parseRawData: parsed "+str(len(cellDataL))+" cells")
+                try:
+                    cellNumber, cellData = parseCellData("\n".join(s))
+                    parsedCellData[cellNumber] = cellData
+                except Exception as e:
+                    logger.error('unexpected error occurred while parsing cellDataL {}'
+                                 .format(rawdata), exception=e)
```

Comment:
Yes, because `parseRawData` has 2 for loops which might explode. I've wrapped all three cases. [parseRawData [for1] [for2] ]